pkgs/development/compilers/scryer-prolog/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, fetchFromGitHub
+, rustPlatform
+, gmp
+, libmpc
+, mpfr
+, openssl
+, pkg-config
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "scryer-prolog";
+  version = "0.8.127";
+
+  src = fetchFromGitHub {
+    owner = "mthom";
+    repo = "scryer-prolog";
+    rev = "v${version}";
+    sha256 = "0307yclslkdx6f0h0101a3da47rhz6qizf4i8q8rjh4id8wpdsn8";
+  };
+
+  cargoSha256 = "0pc8j7b3mpn0q11mqx7icrqfrh05lvik4xq43qpgk8w7lf59hqqy";
+  cargoPatches = [ ./cargo.patch ];
```

Comment:
Can you add a comment here about why the patch is needed, hopefully with a reference to an upstream PR/issue?

Reply:
Add a comment. There is no upstream issue. By default, upstream uses some vendored packages, we are using system packages.